File: src/SOP8/Egovframework/Base/base-utility/src/main/java/egovframework/base/excelreport/models/SheetData.java
package egovframework.base.excelreport.models;

import org.apache.poi.ss.usermodel.CellStyle;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class SheetData {

    private String sheetName = "";
    // 첫 번째 행의 데이터들
    // Key : ColumnIndex
    private Map<Integer, String> titles = new HashMap<>();
    private Map<Integer, CellStyle> titleStyles = new HashMap<>();
    // Key : ColumnIndex
    private Map<Integer, List<String>> columnDatas = new HashMap<>();
    // Value.Key : ColumnData Index
    private Map<Integer, Map<Integer, CellStyle>> cellStyles = new HashMap<>();
    private Integer titleRowHeight = null;
    // Key : ColumnIndex
    private Map<Integer, Integer> rowHeight = new HashMap<>();
    private Object tag = null;

    // Getter / Setter
    public String getSheetName() {
        return sheetName;
    }

    public void setSheetName(String sheetName) {
        this.sheetName = sheetName;
    }

    public Map<Integer, String> getTitles() {
        return titles;
    }

    public Map<Integer, CellStyle> getTitleStyles() {
        return titleStyles;
    }

    public Map<Integer, List<String>> getColumnDatas() {
        return columnDatas;
    }

    public Map<Integer, Map<Integer, CellStyle>> getCellStyles() {
        return cellStyles;
    }

    public Integer getTitleRowHeight() {
        return titleRowHeight;
    }

    public void setTitleRowHeight(Integer titleRowHeight) {
        this.titleRowHeight = titleRowHeight;
    }

    public Map<Integer, Integer> getRowHeight() {
        return rowHeight;
    }

    public Object getTag() {
        return tag;
    }

    public void setTag(Object tag) {
        this.tag = tag;
    }

    // 생성자
    public SheetData(String sheetName) {
        this.sheetName = sheetName;
    }
}

File: src/SOP8/Utility/dnsExcelReport/Models/SheetData.cs
using System;
using System.Collections.Generic;
using NPOI.SS.UserModel;

namespace dnsExcelReport.Models
{
    public class SheetData : IComparable
    {
        private string m_strSheetName = "";
        // 첫번째 행의 데이터들
        // Key : ColumnIndex
        private Dictionary<int, string> m_dicTitles = new Dictionary<int, string>();
        private Dictionary<int, ICellStyle> m_dicTitleStyles = new Dictionary<int, ICellStyle>();
        // Key : ColumnIndex
        private Dictionary<int, List<string>> m_dicColumnDatas = new Dictionary<int, List<string>>();
        // Value.Key : ColumnData Index
        private Dictionary<int, Dictionary<int, ICellStyle>> m_dicCellStyles = new Dictionary<int, Dictionary<int, ICellStyle>>();
        private int? m_titleRowHeight = null;
        // Key : ColumnIndex
        private Dictionary<int, int> m_dicRowHeight = new Dictionary<int, int>();
        private object m_tag = null;

        public string SheetName
        {
            get { return m_strSheetName; }
            set { m_strSheetName = value; }
        }

        // 첫번째 행의 데이터들
        // Key : ColumnIndex
        public Dictionary<int, string> Titles
        {
            get { return m_dicTitles; }
        }

        // Key : ColumnIndex
        public Dictionary<int, ICellStyle> TitleStyles
        {
            get { return m_dicTitleStyles; }
        }

        // Key : ColumnIndex
        public Dictionary<int, List<string>> ColumnDatas
        {
            get { return m_dicColumnDatas; }
        }

        // Key : ColumnIndex
        // Value.Key : ColumnData Index
        public Dictionary<int, Dictionary<int, ICellStyle>> CellStyles
        {
            get { return m_dicCellStyles; }
        }

        public int? TitleRowHeight
        {
            get { return m_titleRowHeight; }
            set { m_titleRowHeight = value; }
        }

        // Key : ColumnIndex
        public Dictionary<int, int> RowHeight
        {
            get { return m_dicRowHeight; }
        }

        public object Tag
        {
            get { return m_tag; }
            set { m_tag = value; }
        }

        public SheetData(string strSheetName)
        {
            m_strSheetName = strSheetName;
        }

        public int CompareTo(object obj)
        {
            SheetData data1 = this;
            SheetData data2 = (SheetData)obj;
            return data1.m_strSheetName.CompareTo(data2.m_strSheetName);
        }
    }
}
